### PR TITLE
feat: cortex chat/run/models start with new model data structure

### PR DESCRIPTION
**Pull request by vansangpfiev**
## Describe Your Changes

Use `model.list` when run `cortex models start` `cortex chat` and `cortex run`

## Fixes Issues

https://github.com/janhq/cortex.cpp/issues/1300

## Self Checklist

- [ ] Added relevant comments, esp in complex areas
- [ ] Updated docs (for bug fixes / features)
- [ ] Created issues for follow-up changes or refactoring needed

### CODE REVIEW

File: engine/commands/run_cmd.cc [hunk 1]
```diff
@@ -48,28 +49,29 @@ void RunCmd::Exec() {
     }
   }
 
-  config::YamlHandler yaml_handler;
-  yaml_handler.ModelConfigFromFile(
-      file_manager_utils::GetModelsContainerPath().string() + "/" + model_file +
-      ".yaml");
-  auto mc = yaml_handler.GetModelConfig();
+  // TODO(sang) refactor after `cortex pull` done with new data structure
```
Comment (namchuai) [inline engine/commands/run_cmd.cc]: `cortex pull` is done 👯

**Review by dan-menlo — APPROVED** [review]
Lgtm, thank you @vansangpfiev

File: engine/utils/cli_selection_utils.h [hunk 2]
```diff
@@ -20,7 +20,7 @@ inline std::optional<std::string> PrintSelection(
   std::string selection{""};
   PrintMenu(options);
   std::cout << "Select an option (" << 1 << "-" << options.size() << "): ";
-  std::cin >> selection;
+  std::getline(std::cin, selection);
```
Comment (vansangpfiev) [inline engine/utils/cli_selection_utils.h:23]: use std::getline to fix issue with interactive shell on `chat`
https://stackoverflow.com/questions/6819082/why-does-a-stdgetline-call-on-stdcin-not-wait-for-user-input

**Review by nguyenhoangthuan99 — APPROVED** [review]

**Review by namchuai — APPROVED** [review]
✅ lgtm

**Pull request merged**